### PR TITLE
#169: Add caching for HTTP registry

### LINKED ISSUE #169: Investigate why listing extensions is slow
We already improved the speed of listing extensions in https://github.com/exasol/extension-manager/issues/147 but it still is too slow. We need to investigate the root cause.

### PR DESCRIPTION
**Pull request by kaklakariada**
Closes #169

This enables caching for the HTTP registry to avoid requesting the same URL multiple times when listing extensions. The PR also cleans up log messages so it's easier to identify parts of the code that are too slow.

### CODE REVIEW

**Comment by pedroscaff** [comment]
@kaklakariada can you please add a short description of what exactly you did/want to achieve? otherwise I'm going through many files but without knowing the exact goal of some changes

File: doc/developer_guide.md [hunk 1]
```diff
@@ -74,9 +76,29 @@ go test -p 1 -short ./...
 
 Please note that also `-short` tests need `-p 1` because extension integration tests share a directory for building a test extension. Tests will fail randomly without `-p 1`.
 
-## Static Code Analysis
+#### Manual Integration Tests With SaaS
```
Comment (pedroscaff) [inline doc/developer_guide.md:79]: This repository is public, right? I'm thinking how much sense this makes outside our organization or if we should have it as internal docs

File: pkg/extensionController/registry/http.go [hunk 2]
```diff
@@ -23,25 +26,38 @@ type httpRegistry struct {
...
 	if err != nil {
 		return nil, err
 	}
-	return h.index.GetExtensionIDs(), nil
+	return index.GetExtensionIDs(), nil
 }
 
-func (h *httpRegistry) loadIndex() error {
-	response, err := getResponse(h.url)
+/* [impl -> dsn~extension-registry.cache~1]. */
```
Comment (pedroscaff) [inline pkg/extensionController/registry/http.go:36]: one question, does this comment formatting has a special meaning?

**Review by pedroscaff — APPROVED** [review]
code-wise approved, one question regarding the documentation. the other questions are for my understanding only

File: doc/developer_guide.md (hunk 1 above)
Comment (kaklakariada) [reply doc/developer_guide.md:79]: You are right. But this does not reveal any secrets. Having separate internal docs will make it only confusing for developers.

File: pkg/extensionController/registry/http.go (hunk 2 above)
Comment (kaklakariada) [reply pkg/extensionController/registry/http.go:36]: This comment is used for requirement tracing [OpenFastTrace](https://github.com/itsallcode/openfasttrace). It marks the specification item `dsn~extension-registry.cache~1` to be implemented in this place in the code.

**Review by pj-spoelders — APPROVED** [review]

**Pull request merged**